From b4d3dd3fb6b44c614836f1fc4aa4f087f4919d69 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Mon, 13 Nov 2017 08:09:16 +0100
Subject: lpfc: check for valid scsi cmnd in lpfc_scsi_cmd_iocb_cmpl()
References: bsc#1051133
Patch-Mainline: submitted linux-scsi 2017/11/09

When the 'host_scribble' backpointer doesn't match we need to
skip processing this command.


Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/lpfc/lpfc_scsi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/scsi/lpfc/lpfc_scsi.c b/drivers/scsi/lpfc/lpfc_scsi.c
index 59c59ed..519cc5a 100644
--- a/drivers/scsi/lpfc/lpfc_scsi.c
+++ b/drivers/scsi/lpfc/lpfc_scsi.c
@@ -4001,6 +4001,10 @@ lpfc_scsi_cmd_iocb_cmpl(struct lpfc_hba *phba, struct lpfc_iocbq *pIocbIn,
 		return;
 	}
 	cmd = lpfc_cmd->pCmd;
+	if (cmd->host_scribble != lpfc_cmd) {
+		spin_unlock_irqrestore(&phba->hbalock, flags);
+		return;
+	}
 	cmd->host_scribble = NULL;
 	spin_unlock_irqrestore(&phba->hbalock, flags);
 
-- 
1.8.5.6

